cpp/ql/src/Security/CWE/CWE-170/ImproperNullTerminationTainted.ql

```diff
@@ -86,4 +86,5 @@ class TaintSource extends VariableAccess {
 
 from TaintSource source, VariableAccess sink
 where source.reaches(sink)
-select sink, "$@ flows to here and may not be null terminated.", source, "User-provided value"
+select sink, "$@ that may not be null terminated flows to a string operation.", source,
```

Comment:
This is a strange phrasing (again the original is not ideal).

Reply:
It's a taint-problem, so using "depends on" might work better.   

I think this could work: 
"String operation depends on a [user-provided value] that may not be null terminated."